File: i2f-jdk/i2f-swl/src/main/java/i2f/swl/cert/SwlCert.java
package i2f.swl.cert;

import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author Ice2Faith
 * @date 2024/8/6 21:44
 * @desc
 */
@Data
@NoArgsConstructor
public class SwlCert {
    protected String certId;
    protected String publicKey;
    protected String privateKey;
    protected String remotePublicKey;

    public SwlCert(String certId, String publicKey, String privateKey, String remotePublicKey) {
        this.certId = certId;
        this.publicKey = publicKey;
        this.privateKey = privateKey;
        this.remotePublicKey = remotePublicKey;
    }

    public static SwlCert copy(SwlCert cert) {
        SwlCert ret = new SwlCert();
        ret.setCertId(cert.getCertId());
        ret.setPublicKey(cert.getPublicKey());
        ret.setPrivateKey(cert.getPrivateKey());
        ret.setRemotePublicKey(cert.getRemotePublicKey());
        return ret;
    }

    public SwlCert copy() {
        return copy(this);
    }
}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/core/cert/data/SwlCert.js
/**
 * @return {SwlCert}
 * @constructor
 */
function SwlCert(certId=null,
                 publicKey=null,
                 privateKey=null,
                 remotePublicKey=null) {
    /**
     * 证书ID
     * @type {String}
     */
    this.certId = certId;
    /**
     * 公钥
     * @type {String}
     */
    this.publicKey = publicKey;
    /**
     * 私钥
     * @type {String}
     */
    this.privateKey = privateKey;
    /**
     * 对方公钥
     * @type {String}
     */
    this.remotePublicKey = remotePublicKey;
}

/**
 *
 * @param cert {SwlCert}
 * @return {SwlCert}
 */
SwlCert.copy = function (cert) {
    let ret = new SwlCert();
    ret.certId=cert.certId;
    ret.publicKey = cert.publicKey;
    ret.privateKey = cert.privateKey;
    ret.remotePublicKey = cert.remotePublicKey;
    return ret;
}

/**
 *
 * @return {SwlCert}
 */
SwlCert.prototype.copy=function(){
    return SwlCert.copy(this)
}

export default SwlCert
